creator-node/src/middlewares/premiumContent/premiumContentMiddleware.ts

```diff
@@ -0,0 +1,125 @@
+import {
+  sendResponse,
+  errorResponseServerError,
+  errorResponseForbidden,
+  errorResponseBadRequest
+} from '../../apiHelpers'
+import { recoverWallet } from '../../apiSigning'
+import { NextFunction, Request, Response } from 'express'
+import { isPremiumContentMatch } from '../../premiumContent/helpers'
+import { PremiumContentType } from '../../premiumContent/types'
+import { getAllRegisteredDNodes } from '../../utils'
+import type Logger from 'bunyan'
+import { Redis } from 'ioredis'
+
+/**
+ * Middleware to validate requests to get premium content.
+ */
+export const premiumContentMiddleware = async (
```

Comment:
The logic in this function is very tightly coupled with the request handling logic/middleware - it would be good if we could pull this out into a standalone function that would handle the signatures and return success or errors, so we could test that without having to mock out req/res etc for middleware

Reply:
going to address in another PR